on_node_configured.sh

```diff
@@ -0,0 +1,219 @@
+
+function modify_slurm_conf() {
+    # add JWT auth and accounting config to slurm.conf
+    # /opt/slurm is shared via nfs, so this only needs to be configured on head node
+
+    cat >>/opt/slurm/etc/slurm.conf <<EOF
+# Enable jwt auth for Slurmrestd
+AuthAltTypes=auth/jwt
+# ACCOUNTING
+JobAcctGatherType=jobacct_gather/linux
+AccountingStorageType=accounting_storage/slurmdbd
+AccountingStorageHost=localhost
+AccountingStoragePort=6819
+EOF
+}
+
+function configure_users_common() {
+    sysctl user.max_user_namespaces=15000
+    usermod --add-subuids 165536-231071 --add-subgids 165536-231071 slurm
+}
+
+function configure_users_head_node() {
+    configure_users_common
+
+    cat <<'EOF' | tee -a /home/centos/.bashrc /home/slurm/.bashrc
+# Set variables to avoid podman conflicts between nodes due to nfs-sharing of /home
+# See basedir-spec at https://specifications.freedesktop.org/
+
+base_xdg_dir=$(mktemp -qd /tmp/"$(id -u)"-xdg-XXXXXXXXXX)
+
+export XDG_RUNTIME_DIR="$base_xdg_dir"/.runtime
+export XDG_DATA_HOME="$base_xdg_dir"/.data
+export XDG_STATE_HOME="$base_xdg_dir"/.state
+export XDG_CACHE_HOME="$base_xdg_dir"/.cache
+export XDG_CONFIG_HOME="$base_xdg_dir"/.config
+
+unset base_xdg_dir
+
+for directory in {"$XDG_RUNTIME_DIR","$XDG_DATA_HOME","$XDG_STATE_HOME","$XDG_CACHE_HOME","$XDG_CONFIG_HOME"}; do
+    mkdir -p "$directory"
+done
+
+alias podman="podman --runroot=\$XDG_RUNTIME_DIR --root=\$XDG_DATA_HOME"
+EOF
+}
+
+function write_jwt_key_file() {
+    # set the jwt key
+    if [ "$SLURM_JWT_KEY" ]; then
+        echo "- JWT secret variable found, writing..."
+
+        mkdir -p "$JWT_KEY_DIR"
+
+        echo -n "$SLURM_JWT_KEY" >"$JWT_KEY_FILE"
+    else
+        echo "Error: JWT key not present in environment - aborting cluster deployment" >&2
+        return 1
+    fi
+
+    chown slurm:slurm "$JWT_KEY_FILE"
+    chmod 0600 "$JWT_KEY_FILE"
+}
+
+function create_slurmrest_conf() {
+    # create the slurmrestd.conf file
+    # this file can be owned by root, because the slurmrestd service is run by root
+    cat >/opt/slurm/etc/slurmrestd.conf <<EOF
+include /opt/slurm/etc/slurm.conf
+AuthType=auth/jwt
+EOF
+
+}
+
+function create_slurmdb_conf() {
+    local slurmdbd_password
+    slurmdbd_password=$(cat "$SLURM_PASSWORD_FILE")
+    # create the slurmdbd.conf file
+    cat >/opt/slurm/etc/slurmdbd.conf <<EOF
+AuthType=auth/munge
+DbdHost=localhost
+DebugLevel=info
+DbdPort=6819
+SlurmUser=slurm
+LogFile=/var/log/slurmdbd.log
+StorageType=accounting_storage/mysql
+StorageUser=${SLURMDBD_USER}
+StoragePass=${slurmdbd_password}
+StorageHost=localhost
+AuthAltTypes=auth/jwt
+AuthAltParameters=jwt_key=${JWT_KEY_FILE}
+EOF
+
+    chown slurm:slurm /opt/slurm/etc/slurmdbd.conf
+    chmod 600 /opt/slurm/etc/slurmdbd.conf
+}
+
+function create_slurmrest_service() {
+
+    cat >/etc/sysconfig/slurmrestd <<'EOF'
+SLURMRESTD_OPTIONS="-a rest_auth/jwt -s openapi/v0.0.37 -u slurmrestd -g slurmrestd 0.0.0.0:8082 -vvvvvvvvvv"
```

Comment:
That's a lotta vs

Reply:
Yeah - 1-10 Vs are valid in Slurm settings! In practice, their debug levels don't seem to go past 5

Reply:
Tbh once you go past 3 I think you probably need to re-think your verbosity specification lol